extensions/workflowEngineDataSource/WorkflowEngineDataSourceQueuePopulator.js

```diff
@@ -135,6 +150,56 @@ class WorkflowEngineDataSourceQueuePopulator extends QueuePopulatorExtension {
         this._updateFilterDescriptors();
     }
 
+    /**
+     * Update the filter descriptor table acc/to new information
+     *
+     * @param {array} results - array of name/values
+     *
+     * @return {undefined}
+     */
+    _processFilterDescriptors(results) {
+        this.log.debug('WEDSQP: processing filter descriptors');
+        // track missing children
+        const _children = {};
+        // create entries / update refcounts
+        Object.values(results).forEach(kv => {
```

Comment:
Since `results` is an array, you can do `results.forEach` directly.

On the other hand, it might make sense to have indeed an object with mapping `key => value` instead of pairs of `[key,value]` stored as array items (mostly to reduce code verbosity)

Reply:
Note that I use Object.values() and I do not master the format of results.